### PR TITLE
Change default coding style to PSR-12

### PR DESCRIPTION
**Pull request by jaylinski**
* Dropped support for phpspec v4
* Updated deps

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
 All notable changes to this project will be documented in this file.
 
+## [v5.0.0] - 2019-11-12
+### Changed
+- Default coding style is now PSR-12
+- Default code style fixer is now `squizlabs/php_codesniffer`
+### Added 
+- Support for phpspec v5 and v6
+### Removed
+- Support for phpspec v4
```
Comment (jaylinski) [inline CHANGELOG.md:11]: This is still a TODO and depends on https://github.com/karriereat/phpspec-matchers/pull/14.

File: src/CodeStyleFixer.php [hunk 2]
```diff
@@ -16,7 +16,7 @@ class CodeStyleFixer implements ComposerScriptInterface
      *
      * @var string
      */
-    private static $command = 'php-cs-fixer fix src --rules=@PSR2 --using-cache=false';
+    private static $command = 'phpcbf src --standard=PSR12';
```
Comment (jaylinski) [inline src/CodeStyleFixer.php:19]: `php-cs-fixer` doesn't support PSR-12 yet: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4502

File: README.md [hunk 3]
```diff
@@ -84,6 +74,7 @@ Usage:
 Options:
       --fail     Exit with 1 if tests fail.
       --notty    Disable TTY.
+      --ptimeout Set process timeout (defaults to 60 seconds).
```
Comment (jaylinski) [inline README.md:77]: Do this in a separate commit.

**Comment by jaylinski** [comment]
I tried to change the `.styleci.yml` preset to `preset: psr12`, but it seems it isn't supported yet.

See also https://twitter.com/GrahamJCampbell/status/1161012976499011585.

**Review by Witti — APPROVED** [review]
